src/EA.Iws.DataAccess/Repositories/NotificationAuditRepository.cs

```diff
@@ -31,7 +31,40 @@ public async Task<IEnumerable<Audit>> GetNotificationAuditsById(Guid notificatio
         {
             await notificationApplicationAuthorization.EnsureAccessAsync(notificationId);
 
-            return await this.context.NotificationAudit.Where(p => p.NotificationId == notificationId).ToArrayAsync();
+            return await this.context.NotificationAudit
+                .Where(p => p.NotificationId == notificationId)
+                .ToArrayAsync();
+        }
+
+        public async Task<IEnumerable<Audit>> GetPagedNotificationAuditsById(Guid notificationId, int pageNumber, int pageSize)
+        {
+            await notificationApplicationAuthorization.EnsureAccessAsync(notificationId);
+
+            IEnumerable<Audit> retval = new List<Audit>();
+
+            try
+            {
+                retval = await this.context.NotificationAudit
+                    .Where(p => p.NotificationId == notificationId)
+                    .OrderByDescending(x => x.DateAdded)
+                    .Skip((pageNumber - 1) * pageSize)
+                    .Take(pageSize)
+                    .ToArrayAsync();
+            }
+            catch (Exception ex)
+            {
+                string test = ex.ToString();
```

Comment:
Nothing's really happening in this `catch` block. I think you can remove the entire `try catch` and return the query directly, i.e. you won't need the local `retval` variable.

Reply:
Whoops! This was left as part of debugging a problem. I was meant to remove, but obviously forgot! I'll remove this now.